File: app/schemas/usage.py
from pydantic import BaseModel


class UsageSummaryResponse(BaseModel):
    start_date: str | None = None
    end_date: str | None = None
    total_calls: int
    total_input_tokens: int
    total_output_tokens: int
    total_estimated_cost_usd: float
    projected_monthly_cost_usd: float
    active_days: int


class UsageByFeatureItem(BaseModel):
    feature: str
    provider: str
    start_date: str | None = None
    end_date: str | None = None
    total_calls: int
    total_input_tokens: int
    total_output_tokens: int
    total_estimated_cost_usd: float
    projected_monthly_cost_usd: float
    active_days: int

File: frontend/src/types/usage.ts
export interface UsageSummaryResponse {
  start_date: string | null;
  end_date: string | null;
  total_calls: number;
  total_input_tokens: number;
  total_output_tokens: number;
  total_estimated_cost_usd: number;
  projected_monthly_cost_usd: number;
  active_days: number;
}

export interface UsageByFeatureItem {
  feature: string;
  provider: string;
  start_date: string | null;
  end_date: string | null;
  total_calls: number;
  total_input_tokens: number;
  total_output_tokens: number;
  total_estimated_cost_usd: number;
  projected_monthly_cost_usd: number;
  active_days: number;
}
